src/preferences/dialog/dlgprefautodj.cpp

```diff
@@ -135,66 +130,53 @@ void DlgPrefAutoDJ::slotResetToDefaults() {
     RequeueIgnoreTimeEdit->setTime(QTime::fromString(
             "23:59", RequeueIgnoreTimeEdit->displayFormat()));
     RequeueIgnoreCheckBox->setChecked(false);
-    m_pConfig->set(ConfigKey("[Auto DJ]", "UseIgnoreTimeBuff"),QString("0"));
+    m_pConfig->setValue(ConfigKey("[Auto DJ]", "UseIgnoreTimeBuff"), false);
     RequeueIgnoreTimeEdit->setEnabled(false);
 
     RandomQueueMinimumSpinBox->setValue(5);
     RandomQueueCheckBox->setChecked(false);
-    m_pConfig->set(ConfigKey("[Auto DJ]", "EnableRandomQueueBuff"),QString("0"));
+    m_pConfig->setValue(ConfigKey("[Auto DJ]", "EnableRandomQueueBuff"), false);
     RandomQueueMinimumSpinBox->setEnabled(false);
     RandomQueueCheckBox->setEnabled(true);
 }
 
 void DlgPrefAutoDJ::slotSetMinimumAvailable(int a_iValue) {
-    QString str;
-    str.setNum(a_iValue);
-    m_pConfig->set(ConfigKey("[Auto DJ]","MinimumAvailableBuff"),str);
+    m_pConfig->setValue(ConfigKey("[Auto DJ]", "MinimumAvailableBuff"), a_iValue);
 }
 
-void DlgPrefAutoDJ::slotToggleRequeueIgnore(int a_iState) {
-    bool bChecked = (a_iState == Qt::Checked);
-    QString strChecked = (bChecked) ? "1" : "0";
-    m_pConfig->set(ConfigKey("[Auto DJ]", "UseIgnoreTimeBuff"), strChecked);
-    RequeueIgnoreTimeEdit->setEnabled(bChecked);
+void DlgPrefAutoDJ::slotToggleRequeueIgnore(bool checked) {
```

Comment:
connected QCheckBox::stateChanged retuns int, because of three states.

Reply:
Ah yep, oversight. I thought it was bound to `QAbstractButton::clicked`.